e3db/src/main/java/com/tozny/e3db/Client.java

```diff
@@ -181,10 +182,27 @@
  *
  * To verify a signed document, use the {@link #verify(SignedDocument, String)} method. Note that the {@link LocalEncryptedRecord} class
  * implements {@link SignedDocument} and thus always has a signature attached that can be verified.
+ *
+ * <h1>Reading &amp; Writing Large Files<h1>
+ *
+ * E3DB can compress and encrypt files for storage. Files are treated much like records, except the data for the file is
+ * not included inline when downloading a record. Instead, a separate request for each file must be made to the {@link #readFile(UUID, File, ResultHandler)}
+ * method.
+ *
+ * To write a file, use the {@link #writeFile(String, File, Map, ResultHandler)} method. The record type and plaintext meta
+ * arguments behave the same as with normal records. The SDK will compress and encrypt the record in the same directory
+ * that stores the plaintext file and will need at least twice as much free space as the size of the plaintext file. The
+ * temporary encrypted file will always be deleted the upload finishes (or if an error occurs).
+ *
+ * To read a file, use the {@link #readFile(UUID, File, ResultHandler)} method (which is as yet not implemented). The SDK will
+ * download the encrypted file to the same directory as the destination given. It will decrypt and decompress in-place, and write
+ * the result to the destination file. Afterwards, the temporary encrypted file is deleted.
  */
 public class  Client {
+  private static final OkHttpClient anonymousClient;
   private static final ObjectMapper mapper;
   private static final MediaType APPLICATION_JSON = MediaType.parse("application/json");
+  private static final MediaType APPLICATION_OCTET = MediaType.parse("application/octet-stream");
```

Comment:
Great use of mime types! I see a lot of people use text/plain instead, but this is 100% the right choice.